components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialRipple.h"
+
+@interface MDCRippleViewUsesSuperviewShadowLayerAsMaskTests : XCTestCase
+@end
+
+@implementation MDCRippleViewUsesSuperviewShadowLayerAsMaskTests
+
+- (void)testDefaultInitializationHasNoLayerMask {
+  // Given
+  MDCRippleView *rippleView = [[MDCRippleView alloc] init];
+
+  // Then
+  XCTAssertEqual(rippleView.rippleStyle, MDCRippleStyleBounded);
+  XCTAssertNil(rippleView.layer.mask);
+}
+
+- (void)testHasNoLayerMaskWhenParentlessAndRippleStyleSetToBounded {
+  // Given
+  MDCRippleView *rippleView = [[MDCRippleView alloc] init];
+
+  // When
+  rippleView.rippleStyle = MDCRippleStyleBounded;
+
+  // Then
+  XCTAssertNil(rippleView.layer.mask);
+}
+
+- (void)testInheritsParentShadowPathWhenRippleStyleSetToBounded {
+  // Given
+  MDCRippleView *rippleView = [[MDCRippleView alloc] init];
+  UIView *parentView = [[UIView alloc] init];
+  [parentView addSubview:rippleView];
+
+  // When
+  parentView.layer.shadowPath = [UIBezierPath bezierPathWithRect:parentView.bounds].CGPath;
+  rippleView.rippleStyle = MDCRippleStyleBounded;
+
+  // Then
+  XCTAssertNotNil(rippleView.layer.mask);
+  if (![rippleView.layer.mask isKindOfClass:[CAShapeLayer class]]) {
+    XCTFail(@"Layer mask was expected to be a CAShapeLayer, but it was a %@ instead.",
+            NSStringFromClass([rippleView.layer.mask class]));
+  }
+  CAShapeLayer *maskShapeLayer = (CAShapeLayer *)rippleView.layer.mask;
+  XCTAssertTrue(CGPathEqualToPath(maskShapeLayer.path, parentView.layer.shadowPath));
+}
+
+- (void)testDoesNotInheritParentShadowPathWhenRippleStyleSetToUnbounded {
+  // Given
+  MDCRippleView *rippleView = [[MDCRippleView alloc] init];
+  UIView *parentView = [[UIView alloc] init];
+  [parentView addSubview:rippleView];
+
+  // When
+  parentView.layer.shadowPath = [UIBezierPath bezierPathWithRect:parentView.bounds].CGPath;
+  rippleView.rippleStyle = MDCRippleStyleUnbounded;
+
+  // Then
+  XCTAssertNil(rippleView.layer.mask);
+}
+
+- (void)testInheritsParentShadowPathWhenRippleStyleSetToBoundedEvenWhenFrameDoesNotEqualBounds {
```

Comment:
Tip: Name this something like `testIncorrectlyInherits...` and add a `TODO` to track the fix.

Reply:
Done.